### PR TITLE
solve the timeout problem when stop the kyuubiServer

### PR DESCRIPTION
**Pull request by zhang1002**
### _Why are the changes needed?_
When you stop the KyuubiServiceDiscovery, you need to close node in zk and delete localNodePath at first. But when EmbeddedZookeeper is closed before KyuubiServiceDiscovery, the operation that stop kyuubiServer will timeout. 
Now use 'org.apache.hadoop.util.ShutdownHookManager' to close each service by different priority.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala [hunk 1]
```diff
@@ -165,4 +166,15 @@ private[kyuubi] object Utils extends Logging {
...
+
+  /**
+    * Add some operations that you want into ShutdownHook
+    * @param hook
+    * @param priority: 0~100
+    */
+  def addHook(hook: Runnable, priority: Int): Unit = {
+    ShutdownHookManager.get().addShutdownHook(
+      hook, priority
+    )
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala:178]: inline these

File: kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala [hunk 2]
```diff
@@ -123,8 +123,7 @@ abstract class ServiceDiscovery private (
   }
 
   // close the EPHEMERAL_SEQUENTIAL node in zk
-  protected def closeServiceNode(): Unit = {
-    if (_serviceNode != null) {
+  protected def closeServiceNode(): Unit = {if (_serviceNode != null) {
```
Comment (yaooqinn) [inline kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala:126]: unnecessary change

File: kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala [hunk 3]
```diff
@@ -165,4 +166,15 @@ private[kyuubi] object Utils extends Logging {
   def isTesting: Boolean = {
     System.getProperty(IS_TESTING.key) != null
   }
+
+  /**
+    * Add some operations that you want into ShutdownHook
+    * @param hook
+    * @param priority: 0~100
+    */
+  def addHook(hook: Runnable, priority: Int): Unit = {
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala:175]: rename it to addShutdownHook to be more specfic

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.2.0
